### DISCUSSION TITLE
If dark:[variable], tailwindcss does not work.

### DISCUSSION THREAD

**Discussion by pruge** [tailwindlabs/tailwindcss#10441] [Help] [answered]
```js
function App() {
  // worked
  const getClassName1 = () => {
    const t = "dark:bg-sky-400";
    return "m-4 rounded-md px-4 py-1 " + t;
  };

  // not worked
  const getClassName2 = () => {
    const t = "bg-sky-500";
    return "m-4 rounded-md px-4 py-1 dark:" + t;
  };

    // worked
  const getClassName3 = (t) => {
    return "m-4 rounded-md px-4 py-1 " + t;
  };

  // not worked
  const getClassName4 = (t) => {
    return "m-4 rounded-md px-4 py-1 dark:" + t;
  };

  return (
    <div className="dark">
      <div className="bg-pink-500 text-white w-screen h-screen">
        {/* worked */}
        <div className={getClassName1()}>world</div>
        {/* not worked */}
        <div className={getClassName2()}>world</div>
        {/* worked */}
        <div className={getClassName3("dark:bg-sky-200")}>world</div>
        {/* not worked */}
        <div className={getClassName4("bg-sky-300")}>world</div>
      </div>
    </div>
  );
}
```

[demo](https://codesandbox.io/p/sandbox/romantic-architecture-0wt0gs?file=%2Fsrc%2FApp.js&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A29%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A27%7D%5D)

If you attach the prefix (dark:), it does not work.
Is there any way to use it like this?

**Comment by andymccray** [comment, marked as answer]
[Per the docs](https://tailwindcss.com/docs/content-configuration#dynamic-class-names), Tailwind will `only find classes that exist as complete unbroken strings in your source files`. In the examples you’ve marked as not working the `dark` keyword is being dynamically prepended to your class name variable. In the first (working) example you’ve got the complete unbroken string with the `dark` prefix. Tailwind doesn’t parse the executed JS, it literally just reads it as if it were a flat file.
